src/components/utils/include/utils/timer.h

```diff
@@ -29,105 +29,116 @@
- * Time setups in ::Start(uint,bool) method and starts time out steps.
- * User can call Start, Stop, IsRunning, GetTimeout.
+ * @brief Timer calls custom callback function after
+ * specified timeout has been elapsed
  */
 class Timer {
  public:
   /**
-   * @brief constructor
-   * @param name for indentify of current timer
-   * @param task_for_tracking is SPtr to trackable task
+   * @brief Constructor
+   * Does not starts timer
+   * @param name Timer's name for identity
+   * @param task Task for tracking
    */
-  Timer(const std::string& name, const TimerTask* task_for_tracking);
+  Timer(const std::string& name, TimerTask* task);
 
   /**
-   * @brief destructor - if timer running : call stop in the body
+   * @brief Destructor
+   * Stops timer if it's running
    */
   ~Timer();
+  
   /**
-   * @brief starts timer with new timeout
-   * @param timeout - time to call method from trackable class
-   * @param repeatable - should timer repeat after calling callback, or no
+   * @brief Starts timer with specified timeout
+   * @param timeout Timer timeout
+   * @param single_shot Shows needs to restart timer after timeout
    */
-  void Start(const Milliseconds timeout, const bool repeatable);
+  void Start(const Milliseconds timeout, const bool single_shot);
+
   /**
-   * @brief stops timer without calling callback
+   * @brief Stops timer if it's running
    */
   void Stop();
+
   /**
-   * @brief method which know about Timer state: is running or not
-   * @return true when timer runned, false when timer stand
+   * @brief Gets current timer status
+   * @return True in case of timer is running, false otherwise
    */
-  bool IsRunning() const;
+  bool is_running() const;
+
   /**
-   * @brief GetTimeout
-   * @return returns timeout
+   * @brief Gets current timer timeout
+   * @return Current timeout in milliseconds.
+   * Null if timer has not been started
    */
-  Milliseconds GetTimeout() const;
+  Milliseconds timeout() const;
 
  private:
-  const std::string name_;
-  const TimerTask* task_;
-  bool repeatable_;
-  uint32_t timeout_ms_;
-  bool is_running_;
-  timer_t timer_;
-  mutable sync_primitives::Lock lock_;
-  sync_primitives::Lock task_lock_;
-
   /**
-   * @brief method called from friend handler_wrapper and call run() from task.
+   * @brief Delegate for timer thread
    */
-  void OnTimeout();
+  class TimerDelegate : public threads::ThreadDelegate {
+   public:
+    /**
+     * @brief Constructor
+     * @param timer Timer instance pointer for callback calling
+     */
+    explicit TimerDelegate(const Timer* timer);
 
-  /**
-   * @brief method for setting correct timeout.
-   * @param timeout - if it`s value = 0, timeout will be setted to 1
-   * There would be no way to stop thread if timeout in lopper will be 0
-   * and if we puts to timer_create zero timeout then we get sys error(22)
-   */
-  void SetTimeoutUnsafe(const Milliseconds timeout);
+    void set_timeout(const Milliseconds timeout);
```

Comment:
MAybe also provide doxygen?
For example, Is this time includes `OntImeout` logics routine?
What will be in case of 0 or maximum uint32 value?